subprojects/shipkit/src/main/groovy/org/shipkit/internal/gradle/javadoc/JavadocPlugin.java

```diff
@@ -95,8 +96,12 @@ public void apply(Project project) {
         GitPushTask pushJavadoc = createPushJavadocTask(project, conf);
         pushJavadoc.dependsOn(commitJavadocTask);
 
-        Task releaseJavadocTask = createReleaseJavadocTask(project, cloneJavadocTask, commitJavadocTask, pushJavadoc);
+        Task releaseJavadocTask = createReleaseJavadocTask(project);
         releaseJavadocTask.dependsOn(cloneJavadocTask, commitJavadocTask, pushJavadoc);
+        Task performReleaseTask = project.getRootProject().getTasks().findByName(PERFORM_RELEASE_TASK);
```

Comment:
I suggest to use the following to avoid ordering issues when folks declare "javadoc" or "shipkit" plugin in different order (pseudo code):

```
project.plugins.withType(PluginThatAddsPerformRelease) {
  tasks.performRelease.dependsOn ...
}
```

Reply:
Good catch. It's easy to reproduce in `ShipkitJavadocIntegTest` by exchange 34 & 35 lines. 
But I don't know how to write it in Java code :-(